R/qenv-get_code.R

```diff
@@ -11,7 +12,7 @@
 #' get_code(q3)
 #' get_code(q3, deparse = FALSE)
 #' @export
-setGeneric("get_code", function(object, deparse = TRUE) {
+setGeneric("get_code", function(object, deparse = TRUE, names = NULL) {
```

Comment:
I don't like the argument name `names`. I would prefer `object`, `target`, `dataset`.

Actually, there is no validation that the content of `names` matches any name tagged with `@effect`, is there? Should there be? Do we need another slot to store tagged names? But then we would have to parse dependencies before when code is added, not retrieved.


Btw. I think a better tag name would be `@affetcs`.

Reply:
agree that `@affects` is a better name. names passed in `@effect`/`@affects` are stored in `code_dependency()@effects`.
I don't think there should be anything matching `names` and stuff tagged with `@effect` as the object passed in `names` does not need to be tagged with `@effect`. `names` are checked for the existence in `object@env` only.

for `names` I would prefer `object` but this is already a parameter that corresponds to `qenv`. Agree that we can come up with something better

Reply:
hey, moved both questions to the main thread https://github.com/insightsengineering/teal.code/pull/146#issuecomment-1746815397